### PR TITLE
Migrate to pnpm

### PR DESCRIPTION
**Pull request by M6AI**

### CODE REVIEW

File: testplan/web_ui/testing/package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "pegjs": "^0.11.0-master.b7b87ea",
     "plotly.js": "^1.58.5",
     "react": "16.13.1",
+    "react-app-polyfill": "1.0.6",
```
Comment (M6AI) [inline testplan/web_ui/testing/package.json:35]: It appears that NPM/Yarn flattens the hierarchy, but PNPM uses links in the background and every explicit import needs to be present in dependencies.

File: testplan/web_ui/testing/package.json [hunk 2]
```diff
@@ -43,7 +44,7 @@
     "react-scripts": "^3.4.1",
     "react-spinners": "^0.6.0",
     "react-syntax-highlighter": "^11.0.2",
-    "react-test-renderer": "16.6.0",
+    "react-test-renderer": "16.13.1",
```
Comment (M6AI) [inline testplan/web_ui/testing/package.json:47]: Enables shallow from enzyme to work with useState hook.

File: install-testplan-ui [hunk 3]
```diff
@@ -92,9 +102,18 @@ def install_and_build_ui(path=TESTPLAN_UI_DIR, dev=False, verbose=False):
...
+            shell=True,
+            cwd=path,
+            stdout=output,
+            stderr=subprocess.STDOUT,
+        )
+
+        logging.info("Enabling pre and post scripts...")
         subprocess.check_call(
-            "yarn install {} --network-timeout 600000".format(production_cmd),
+            "pnpm config set enable-pre-post-scripts true",
```
Comment (M6AI) [inline install-testplan-ui:116]: PNPM does not enable pre and post scripts by default and the local config needs to be updated.

Comment (yuxuan-ms) [reply install-testplan-ui:116]: I guess this command would fail on pnpm5?

**Review by yuxuan-ms — APPROVED** [review]

**Review by yuxuan-ms — APPROVED** [review]

**Review by yuxuan-ms — APPROVED** [review]

**Pull request merged**